knotx-knots/knotx-knot-api/src/main/java/com/cognifide/knotx/knot/api/AbstractKnot.java

```diff
@@ -48,24 +53,51 @@ public void init(Vertx vertx, Context context) {
   public void start() throws Exception {
     LOGGER.info("Starting <{}>", this.getClass().getName());
 
-    vertx.eventBus().<KnotContext>consumer(configuration.getAddress())
-        .handler(message -> Observable.just(message)
-            .doOnNext(this::traceMessage)
-            .subscribe(
-                result -> process(result, message::reply),
-                error -> {
-                  LOGGER.error("Error occurred in " + this.getClass().getName() + ".", error);
-                  message.reply(processError(message.body(), error));
-                }
-            ));
+    Observable<Message<KnotContext>> observable = vertx.eventBus().<KnotContext>consumer(
+        configuration.getAddress()).toObservable();
+    observable
+        .doOnNext(this::traceMessage)
+        .subscribe(
+            msg -> {
+              if (shouldProcess(msg)) {
+                process(msg.body())
+                    .subscribe(
+                        msg::reply,
+                        error -> {
+                          LOGGER
+                              .error("Error occurred in " + this.getClass().getName() + ".", error);
+                          msg.reply(processError(msg.body(), error));
+                        });
+              } else {
+                msg.reply(msg.body());
+              }
+            }
+        );
   }
 
-  protected abstract void process(Message<KnotContext> message, Handler<KnotContext> handler);
+  protected abstract rx.Observable<KnotContext> process(KnotContext message);
+
+  protected abstract boolean shouldProcess(Set<String> fragmentsIdentifiers);
 
   protected abstract KnotContext processError(KnotContext knotContext, Throwable error);
 
   protected abstract C initConfiguration(JsonObject config);
 
+  protected boolean shouldProcess(Message<KnotContext> msg) {
+    Optional<List<Fragment>> fragments = msg.body().fragments();
+    Set<String> allFragmentsIdentifiers = getAllFragmentsIdentifiers(fragments);
```

Comment:
`allFragmentsIdentifiers` to `allFragmentsKnotTypes`
`getAllFragmtntsIdentifiers()` to `getAllFragmentsKnotTypes()`..